includes/Core/Storage/User_Setting.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Storage\User_Setting
+ *
+ * @package   Google\Site_Kit\Core\Storage
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Storage;
+
+/**
+ * Base class for a single user setting.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+abstract class User_Setting {
+
+	/**
+	 * The user option name for this setting.
+	 * Override in a sub-class.
+	 */
+	const OPTION = '';
+
+	/**
+	 * User_Options_Interface implementation.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Options_Interface
+	 */
+	protected $user_options;
+
+	/**
+	 * User_Setting constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param User_Options_Interface $user_options User_Options_Interface instance.
+	 */
+	public function __construct( User_Options_Interface $user_options ) {
+		$this->user_options = $user_options;
+	}
+
+	/**
+	 * Registers the setting in WordPress.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		register_meta(
+			'user',
+			$this->user_options->get_meta_key( static::OPTION ),
+			array(
+				'type'              => $this->get_type(),
+				'sanitize_callback' => $this->get_sanitize_callback(),
+				'show_in_rest'      => $this->get_show_in_rest(),
```

Comment:
Let's not introduce the same convenience layer for `show_in_rest` (yet). We don't have it for options either, and it's at this point just one specific case where we need it - it's uncertain we'll ever need it.

The `Tracking` / `Tracking_Optin` user option could override this method and pass `show_in_rest` as `true`.